src/tasks/mtm.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Masked Token Modeling (MTM) Task.
+Include Masked Language Modeling (MLM) and Masked Patch Prediction (MPP).
+"""
+
+from typing import Mapping
+
+import dataclasses
+import tensorflow as tf
+
+from official.core import base_task
+from official.core import config_definitions as cfg
+from official.core import task_factory
+from official.modeling import tf_utils
+from official.nlp.data import data_loader_factory
+from official.nlp.modeling import layers
+
+from configs import encoders
+import input_utils
+from modeling import losses
+from configs import mmt
+from modeling import models
+
+
+@dataclasses.dataclass
+class MaskedTMConfig(cfg.TaskConfig):
+  """The model config."""
+  model: mmt.PretrainModelConfig = mmt.PretrainModelConfig()
+
+  scale_loss: bool = False
+  train_data: cfg.DataConfig = cfg.DataConfig()
+  validation_data: cfg.DataConfig = cfg.DataConfig()
+
+
+@task_factory.register_task_cls(MaskedTMConfig)
+class MaskedTMTask(base_task.Task):
+  """Task object for Masked Tokens Modeling (MTM).
```

Comment:
Some times you say "Masked Token Modeling" and some times "Masked Tokens Modeling", please unify. "Masked Token Modeling" looks like the correct one to me.